### ISSUE TITLE
archive_file doesn't successfully extract zip file if destination is "c:\program files"

### ISSUE THREAD

**Issue by bankulin** [chef/chef#10110]
## Description
archive_file doesn't successfully extract zip file if destination is "c:/program files"

## Chef Version
16.2.73

## Platform Version
Windows 2016 - 1607

## Replication Case
Create a simple recipe to extract a zip file to c:\program files

archive_file 'c:/temp/test.zip' do
  destination "C:/Program Files"
end

## Client Output
       Converging 1 resources
       Recipe: cookbook::default
         * archive_file[c:\temp\test.zip] action extract (up to date)

       Running handlers:
       Running handlers complete
       Chef Infra Client finished, 0/1 resources updated in 06 seconds
       Downloading files from <default-windows2016>
       Finished converging <default-windows2016> (0m21.44s).

However the zip file does not extract.

**Comment by phiggins** [comment]
Hi,

This behavior is by design, because `C:\Program Files` is an existing directory on your system. If I create a simple resource with a destination that already exists and run `chef-apply` with debug logging, I get a message saying that:

```
PS C:\vagrant\chef> bundle exec ruby -Ilib/ .\chef-bin\bin\chef-apply .\test_archive_file_resource.rb --log-level=debug
[2020-07-06T20:25:23+00:00] DEBUG: Reading products and relationships...
[2020-07-06T20:25:23+00:00] DEBUG: Successfully read products and relationships
[2020-07-06T20:25:23+00:00] DEBUG: Searching for the following licenses: ["infra-client", "inspec"]
[2020-07-06T20:25:23+00:00] DEBUG: Found license chef_infra_client at C:/Users/vagrant/.chef/accepted_licenses/chef_infra_client
[2020-07-06T20:25:23+00:00] DEBUG: Found license inspec at C:/Users/vagrant/.chef/accepted_licenses/inspec
[2020-07-06T20:25:23+00:00] DEBUG: Missing licenses remaining: []
[2020-07-06T20:25:23+00:00] DEBUG: All licenses present
[2020-07-06T20:25:23+00:00] DEBUG: Running Ohai with the following configuration: {:logger=>#<Mixlib::Log::Child:0x000000000b55c548 @parent=Chef::Log, @metadata={:system=>"ohai", :version=>"16.2.3"}>}
[2020-07-06T20:25:23+00:00] DEBUG: Running Ohai with the following configuration: {:logger=>#<Mixlib::Log::Child:0x000000000b55c548 @parent=Chef::Log, @metadata={:system=>"ohai", :version=>"16.2.3"}>}
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/win32-process-0.8.3/lib/win32/process.rb:744:in `kill': warning: $SAFE will become a normal global variable in Ruby 3.0 (StructuredWarnings::BuiltInWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/win32-process-0.8.3/lib/win32/process.rb:744:in `kill': warning: $SAFE will become a normal global variable in Ruby 3.0 (StructuredWarnings::BuiltInWarning)
[2020-07-06T20:25:58+00:00] DEBUG: Extracting run list from JSON attributes provided on command line
[2020-07-06T20:25:58+00:00] DEBUG: Applying attributes from json file
[2020-07-06T20:25:58+00:00] DEBUG: Platform is windows version 10.0.17763
[2020-07-06T20:25:58+00:00] INFO: Run List is []
[2020-07-06T20:25:58+00:00] INFO: Run List expands to []
Recipe: (chef-apply cookbook)::(chef-apply recipe)
  * archive_file[test.zip] action extract[2020-07-06T20:25:58+00:00] INFO: Processing archive_file[test.zip] action extract ((chef-apply cookbook)::(chef-apply recipe) line 1)
[2020-07-06T20:25:58+00:00] DEBUG: Not extracting archive as C:\tmp exists and resource not set to overwrite.
 (up to date)
[2020-07-06T20:25:58+00:00] DEBUG: Exiting
```

Adding `overwrite true` to my resource gets the behavior it seems like you're expecting:
```
PS C:\vagrant\chef> bundle exec ruby -Ilib/ .\chef-bin\bin\chef-apply .\test_archive_file_resource.rb --log-level=debug
[2020-07-06T20:27:40+00:00] DEBUG: Reading products and relationships...
[2020-07-06T20:27:40+00:00] DEBUG: Successfully read products and relationships
[2020-07-06T20:27:40+00:00] DEBUG: Searching for the following licenses: ["infra-client", "inspec"]
[2020-07-06T20:27:40+00:00] DEBUG: Found license chef_infra_client at C:/Users/vagrant/.chef/accepted_licenses/chef_infra_client
[2020-07-06T20:27:40+00:00] DEBUG: Found license inspec at C:/Users/vagrant/.chef/accepted_licenses/inspec
[2020-07-06T20:27:40+00:00] DEBUG: Missing licenses remaining: []
[2020-07-06T20:27:40+00:00] DEBUG: All licenses present
[2020-07-06T20:27:40+00:00] DEBUG: Running Ohai with the following configuration: {:logger=>#<Mixlib::Log::Child:0x000000000852e290 @parent=Chef::Log, @metadata={:system=>"ohai", :version=>"16.2.3"}>}
[2020-07-06T20:27:40+00:00] DEBUG: Running Ohai with the following configuration: {:logger=>#<Mixlib::Log::Child:0x000000000852e290 @parent=Chef::Log, @metadata={:system=>"ohai", :version=>"16.2.3"}>}
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/win32-process-0.8.3/lib/win32/process.rb:744:in `kill': warning: $SAFE will become a normal global variable in Ruby 3.0 (StructuredWarnings::BuiltInWarning)
C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/win32-process-0.8.3/lib/win32/process.rb:744:in `kill': warning: $SAFE will become a normal global variable in Ruby 3.0 (StructuredWarnings::BuiltInWarning)
[2020-07-06T20:28:15+00:00] DEBUG: Extracting run list from JSON attributes provided on command line
[2020-07-06T20:28:15+00:00] DEBUG: Applying attributes from json file
[2020-07-06T20:28:15+00:00] DEBUG: Platform is windows version 10.0.17763
[2020-07-06T20:28:15+00:00] INFO: Run List is []
[2020-07-06T20:28:15+00:00] INFO: Run List expands to []
Recipe: (chef-apply cookbook)::(chef-apply recipe)
  * archive_file[test.zip] action extract[2020-07-06T20:28:15+00:00] INFO: Processing archive_file[test.zip] action extract ((chef-apply cookbook)::(chef-apply recipe) line 1)
[2020-07-06T20:28:15+00:00] DEBUG: Overwriting existing content at C:\tmp due to resource's overwrite property settings.

    - extract C:/vagrant/chef/test.zip to C:\tmp
[2020-07-06T20:28:15+00:00] DEBUG: Exiting
```

**Comment by bankulin** [comment]
This is not the behaviour of the windows_zipfile resource in the windows cookbook.

I feel that if chef are wanting us to move off the windows cookbook to using native resources (as indicated with cookstyle failures), then the replacement should have the same behaviour as the resource it is replacing.

**Comment by OceaneLonneux-CAL** [comment]
Hello, will this be looked at? 
We are facing a similar issue, the root cause seems to be the same.

Scenario:
We have this code :
```
archive_file zip_file do
  destination target_dir
  only_if { source_resource.updated_by_last_action? }
end
```
target_dir is a C:/something (that is not C:/Program Files). There is already one folder inside, but it is not linked to when we use this archive_file.
What happens is the following:
- No error message, however, when we check the target_dir, nothing is there,
- I have looked online, and saw that I should use "overwrite true". I have then added it.
- It works fine the first time. However, converging the second time, we get `Can't unlink already-existing object`. The initial file is coming from remote_file, so I added `force_unlink  true` to it, but that didn't change anything.

We did expect archive_file to have the same behaviour than windows_file. Most of the time, archive_file doesn't cause issues, but in this case, we can't run two times in a row that recipe. 

Thank you for your help.
